src/ucp/rndv/rndv.c

```diff
@@ -1814,9 +1817,8 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_rtr_handler,
     }
 
 out_send:
-    /* if it not a PUT pipeline protocol, delete the send request ID */
-    ucp_worker_del_request_id(worker, sreq,
-                              sreq->send.msg_proto.sreq_id);
```

Comment:
where it is overwritten?

Reply:
just asked @dmitrygx offline the same question. let's continue here...

Reply:
yes, looks like this PR does not change anything

Reply:
seems it's overwritten while filling other parts of req->send union (for example sreq->send.rndv_put)

Reply:
right, it can be the case. Was confused by PR description mentioning AM RNDV flow

Reply:
it was overwritten here - https://github.com/dmitrygx/ucx/blob/744a6815dfb29ec7c63698624d40bcc63e714485/src/ucp/rndv/rndv.c#L1794
since `rreq_id` and `sreq_id` are the part of `union`